pkg/store/postings_codec.go

```diff
@@ -82,27 +84,53 @@ func diffVarintEncodeNoHeader(p index.Postings, length int) ([]byte, error) {
 	return buf.B, nil
 }
 
-func diffVarintSnappyDecode(input []byte) (index.Postings, error) {
+var snappyDecodePool sync.Pool
+
+type closeablePostings interface {
+	index.Postings
+	close()
+}
+
+func diffVarintSnappyDecode(input []byte) (closeablePostings, error) {
 	if !isDiffVarintSnappyEncodedPostings(input) {
 		return nil, errors.New("header not found")
 	}
 
-	raw, err := snappy.Decode(nil, input[len(codecHeaderSnappy):])
+	var dstBuf []byte
+	decodeBuf := snappyDecodePool.Get()
+	if decodeBuf != nil {
+		dstBuf = *(decodeBuf.(*[]byte))
+	}
+
+	raw, err := s2.Decode(dstBuf, input[len(codecHeaderSnappy):])
 	if err != nil {
 		return nil, errors.Wrap(err, "snappy decode")
 	}
 
-	return newDiffVarintPostings(raw), nil
+	biggerSlice := raw
+	if cap(dstBuf) > cap(biggerSlice) {
```

Comment:
So this checks if the `s2.Decode` returned a larger slice and frees that, or frees the allocated dstBuf? Maybe we can add a small comment here to clarify what this is doing in relation to `s2.Decode`?

Reply:
Mhm, now that I think about it - we can return both slices to the pool, not just the bigger one. Let me change the code to make it clearer.